timemachine/fe/rbfe.py

```diff
@@ -695,31 +696,52 @@ def estimate_relative_free_energy_bisection_hrex(
 
     try:
         # First phase: bisection to determine lambda spacing
+        md_params_bisection = replace(md_params, n_frames=n_frames_bisection)
         results, trajectories_by_state = run_sims_bisection(
             [lambda_min, lambda_max],
             make_optimized_initial_state,
-            replace(md_params, n_frames=n_frames_bisection),  # TODO: clean up
+            md_params_bisection,
             n_bisections=len(lambda_grid) - 2,
             temperature=temperature,
             min_overlap=min_overlap,
         )
 
         # Second phase: sample initial states determined by bisection using HREX
 
-        # Use equilibrated samples from bisection as initial states
+        def get_mean_final_barostat_volume_scale_factor(trajectories_by_state: Iterable[Trajectory]) -> Optional[float]:
+            scale_factors = [traj.final_barostat_volume_scale_factor for traj in trajectories_by_state]
+            if any(x is not None for x in scale_factors):
+                assert all(x is not None for x in scale_factors)
+                sfs = cast(List[float], scale_factors)  # implied by assertion but required by mypy
+                return float(np.mean(sfs))
+            else:
+                return None
```

Comment:
Should this None be allowed? We disable the scaling and set the value to None, does that mean we just expect to shut off the barostat in this case?

Perhaps we should at least throw a warning as this is likely a non-desirable configuration?

Reply:
I think `None` is valid and expected in the case of a vacuum simulation. It would be invalid if any of the states have a barostat defined, but that should be ruled out in this branch. Will try to think of ways to make this clearer.

Reply:
Added assertions to clarify this in https://github.com/proteneer/timemachine/pull/1151/commits/e56be2b87703906551740bacc368be1c528debb8